cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/GitHubStore.java

```diff
@@ -0,0 +1,94 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.store;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.api.dataset.lib.CloseableIterator;
+import io.cdap.cdap.internal.github.GitHubRepo;
+import io.cdap.cdap.spi.data.StructuredRow;
+import io.cdap.cdap.spi.data.StructuredTable;
+import io.cdap.cdap.spi.data.table.field.Field;
+import io.cdap.cdap.spi.data.table.field.Fields;
+import io.cdap.cdap.spi.data.table.field.Range;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import io.cdap.cdap.store.StoreDefinition;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class GitHubStore {
+  private final TransactionRunner transactionRunner;
+  @Inject
+  public GitHubStore(TransactionRunner transactionRunner) {
+    this.transactionRunner = transactionRunner;
+  }
+
+  public void addOrUpdateRepo(String nickname, String url, String defaultBranch,
+           String authString) throws IOException {
+    TransactionRunners.run(transactionRunner, context ->  {
+      StructuredTable table = context.getTable(StoreDefinition.GitHubStore.GIT_REPOS);
+      Collection<Field<?>> fields = new ArrayList<>();
+      fields.add(Fields.stringField(StoreDefinition.GitHubStore.NICKNAME_FIELD, nickname));
+      fields.add(Fields.stringField(StoreDefinition.GitHubStore.URL_FIELD, url));
+      fields.add(Fields.stringField(StoreDefinition.GitHubStore.DEFAULT_BRANCH_FIELD, defaultBranch));
+      fields.add(Fields.stringField(StoreDefinition.GitHubStore.AUTH_STRING_FIELD, authString));
+      table.upsert(fields);
+    }, IOException.class);
+
+  }
+
+  public void deleteRepo(String nickname) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable table = context.getTable(StoreDefinition.GitHubStore.GIT_REPOS);
+      table.delete(Collections.singleton(Fields.stringField(StoreDefinition.GitHubStore.NICKNAME_FIELD, nickname)));
+    }, IOException.class);
+  }
+
+  public GitHubRepo getRepo(String nickname) throws IOException {
+    return TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable table = context.getTable(StoreDefinition.GitHubStore.GIT_REPOS);
+      Optional<StructuredRow> out = table.read(Collections.singleton(
+          Fields.stringField(StoreDefinition.GitHubStore.NICKNAME_FIELD, nickname)));
+      String nName = out.get().getString(StoreDefinition.GitHubStore.NICKNAME_FIELD);
```

Comment:
instead of directly calling get() , you should check isPresent, other wise you should return null. or your return value of this method should return Optional<GitHubRepo>